include/spi.h

```diff
@@ -141,6 +137,32 @@ struct spi_cs_control {
 	u32_t		delay;
 };
 
+/**
+ * @brief Slave events
+ */
+enum spi_slave_event {
+	/** When slave chip is deselected */
+	SPI_SLAVE_EVENT_DESELECTED	= 0,
+	/** When slave chip is selected */
+	SPI_SLAVE_EVENT_SELECTED,
```

Comment:
It may be difficult, or nearly impossible, to provide these two events on some controllers. nRF5 family will be such case.
Do we really need them?

Reply:
If slave CS is deferred to gpio, it will be useful. Not all slave will need these but some might.

Reply:
If these two events are optional, it should be documented. So that no one expects an event that will never appear.

Reply:
I don't really see how the controller can send the SPI_SLAVE_EVENT_INPUT event ? how do the controller knows that ?
For the SPI_SLAVE_EVENT_ERROR, this should be returned by the transceive call, no ?

Reply:
Documentation is too thin indeed.

About the error event I am not entirely decided either (as for the callback returning an int, see below): let's say there is a driver issue, we could of course return an error code on the spi_transceive() (the very first one which is waiting for sync), and it will actually do that. But what about the slave callback and the slave's state machine logic (if any, some slave will be super dumb, some will be quite complicated): maybe it's worth notifying the error up-front via the callback so slave code could cleanup things etc... 
Let's see, maybe I am over-complexifying the slave user part, and up to the dev to handle the error code only for spi_transceive and cleanup/reset what's necessary.